### PR TITLE
Update DFDs for ETC1S and UASTC

### PR DESCRIPTION
**Pull request by MarkCallow**

### CODE REVIEW

File: ktxspec.adoc [hunk 1]
```diff
@@ -879,59 +879,110 @@ standard gamut images.
...
-or without first conditioning the data with Rate Distortion
-Optimization. If supercompression is used, the DFD's `bytesPlane[0-7]`
-must be set to 0 as described in the next subsection.
-
+Basis Universal UASTC Format::
+The Universal ASTC image format (UASTC) is indicated by `colorModel`
+`KDF_DF_MODEL_UASTC` (= 166) together with `vkFormat` `VK_FORMAT_UNDEFINED`
+(= 0). The DFD must be as described in Section 5.6.14 "KHR_DF_MODEL_UASTC"
+of <<KDF13>>. Images in this format must be transcoded to a
+GPU-supported block-compressed format before being uploaded to and
```
Comment (lexaknyazev) [inline ktxspec.adoc:895]: > Images in this format must be transcoded to a GPU-supported block-compressed format

or decoded to a GPU-supported uncompressed format

File: ktxspec.adoc [hunk 2]
```diff
@@ -879,59 +879,110 @@ standard gamut images.
...
+or `RRR` + `GGG` with `RGB` or `RRR` being the first slice.
++
+TIP: KTX writers may map components of their original input images
+into the RGB and A components of the supercompressed image in any
+way they choose.  They may also offer an option to apply KTXswizzle
+metadata prior to supercompressing an uncompressed KTX2 file.
++
+Images in this format are always are always supercompressed with
+BasisLZ and must be inflated and transcoded to a GPU-supported
+block-compressed format before being uploaded to and sampled by a
```
Comment (lexaknyazev) [inline ktxspec.adoc:964]: Same.

File: ktxspec.adoc [hunk 3]
```diff
@@ -879,59 +879,110 @@ standard gamut images.
...
-For BasisLZ, `texelBlockDimension[0-3]` and each sample's
-`bitlength`, `bitOffset`, `sampleLower` and `sampleUpper` must be
-0
+[[etc1s]]
+Basis Universal ETC1S Format::
+The ETC1S image format is indicated by `colorModel` `KDF_DF_MODEL_ETC1S`
+(= 163) together with `vkFormat` `VK_FORMAT_UNDEFINED` (= 0). The
+DFD must be as described in Section 5.6.11 "KHR_DF_MODEL_ETC1S" of
+<<KDF13>>. Because ETC1S does not support an alpha component, Basis
+Universal uses 2 _slices_, (_planes_ in DFD-speak).
```
Comment (lexaknyazev) [inline ktxspec.adoc:932]: Planes or samples? Also see below.

File: ktxspec.adoc [hunk 4]
```diff
@@ -1288,8 +1296,8 @@ in the <<_level_index,Level Index>>.
 The offset of the start of the alpha slice within the
 <<_levelimages,levelImages>> of its mip level and its byte length.
 
-If `hasAlphaSlices` is unset these values must be 0. If `isETC1s` and
-`hasAlphaSlices` are set these values must be non-zero.
+If the data format descriptor's colorModel is `KHR_DF_MODEL_ETC1S`
+and it has only one sample these values must be 0.
```
Comment (lexaknyazev) [inline ktxspec.adoc:1300]: Sample or plane?

File: ktxspec.adoc (hunk 1 above)
Comment (MarkCallow) [reply ktxspec.adoc]: Done.

File: ktxspec.adoc (hunk 2 above)
Comment (MarkCallow) [reply ktxspec.adoc]: Done.

File: ktxspec.adoc (hunk 3 above)
Comment (MarkCallow) [reply ktxspec.adoc:933]: Slices correspond to planes. Samples are more general. I think this is fine.

File: ktxspec.adoc (hunk 4 above)
Comment (MarkCallow) [reply ktxspec.adoc]: I've updated this to

```
If there is only one slice these values must be 0. For ETC1S with the data
format descriptor colorModel of `KHR_DF_MODEL_ETC1S` this corresponds to
the DFD having only one sample. (As the format is supercompressed
`bytesPlane` fields can't be used to determine the number of planes.)
```

I also added a tip in the ETC1S section.

**Pull request merged**